tsfc/driver.py

```diff
@@ -115,12 +107,7 @@ def compile_integral(integral_data, form_data, prefix, parameters):
     def cellvolume(restriction):
         from ufl import dx
         form = 1 * dx(domain=mesh)
-        fd = compute_form_data(form,
-                               do_apply_function_pullbacks=True,
-                               do_apply_integral_scaling=True,
-                               do_apply_geometry_lowering=True,
-                               do_apply_restrictions=True,
-                               do_estimate_degrees=True)
+        fd = ufl_utils.compute_form_data(form, preserve_geometry_types=())
```

Comment:
Just a note, but whether `CellVolume` and `FacetArea` are specified to be preserved does not matter much. In this case, and similarly for `facetarea` you can see that the form does not contain any of these geometric types. When the mesh geometry is non-affine, UFL will not lower these types anyway. And when UFL lowers these to reference quantities, TSFC still handles the result correctly (though this case has not been tested recently).

The reason I decided to always preserve these geometry types is because

- for some reason I get slightly more common subexpressions eliminated.
- restriction propagation for these types was bugged in UFL, which is worked around by handling them always in TSFC.